### DISCUSSION TITLE
Change in @click.away

### DISCUSSION THREAD

**Discussion by Pab89** [alpinejs/alpine#2083] [7. Bugs (archived)]
Hi, I started noticing some bugs in our code today, when fixating the alpine version to 3.3.3 they went away. So it is 3.3.4 that is introducing the error for us.

I am not sure if the change is a planned change, or it is a bug. But the difference is on @click.away. Earlier when you had @click.away on an element with x-show, @click.away was only triggered if the element was visible. Now it is triggered even if it not visible.

I copied our code below, this code works in 3.3.3, but not in 3.3.4, because if some is clicking on a button outside to $dispatch the "open-mobile-menu" event, the click.away is triggered afterwards and closing the menu, right after it was opened. Personally I think it makes sense to only trigger click.away on visible elements.

```		
<div  x-data="{
			init(){
				},
				open: false,
				currentDisplay: 'frontpage',
				openMenu(){
                    document.documentElement.scrollTop = 0;
					this.currentDisplay = event.detail.startPage ?? 'frontpage';
					this.$dispatch('show-overlay-2');
					this.open = true;
				},
				closeMenu(){
					this.$dispatch('hide-overlay-2');
					this.open = false;
				},
			}"
			class="w-4/5 absolute bg-white z-50 max-w-[17.5rem] min-h-screen"
			x-cloak
			x-show="open"
			x-on:open-mobile-menu.window="openMenu"
			x-on:close-mobile-menu.window="closeMenu"
			@keydown.escape.window="closeMenu"
            @click.away="closeMenu"
		>
Here goes the menu
</div>
```

The above code fails, when click somewhere to open the mobile-menu, because click.away and thereby closeMenu is triggered right afterwards.

**Comment by SimoTod** [comment]
I suspect this is the change that broke the code: https://github.com/alpinejs/alpine/pull/2076/files
The requestAnimationFrame was required to delay the show execution so it would run after the event bubbled up to the top and triggered the away callback. I don't think there is a test for it so it wasn't spotted by cypress.
I think requestAnimationFrame was removed to fix a different bug but the code could use setTimeout instead.
